File: LeetCode problem set/28_Find_the_Index_of_the_First_Occurrenc.cpp
// https://leetcode.com/problems/find-the-index-of-the-first-occurrence-in-a-string/
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int strStr(string haystack, string needle) {
      if(needle.size() > haystack.size()) return -1;
      int len = haystack.size() - needle.size() + 1;
      for (int i = 0; i < len; i++){
        for (int j = 0; j < needle.size(); j++) {
          if(haystack[i+j] != needle[j]) break;
          else if(j== needle.size()-1) return i;
        }
      }
      return -1;
    }
};

int main() {
  Solution s;
  string haystack = "hello";
  string needle = "ll";
  cout << s.strStr(haystack, needle) << endl;
  return 0;
}

File: LeetCode problem set/28_Find_the_Index_of_the_First_Occurrenc.js
// https://leetcode.com/problems/find-the-index-of-the-first-occurrence-in-a-string/
const strStr = (haystack, needle) => {
  if (needle.length > haystack.length) return -1;
  let len = haystack.length - needle.length + 1;
  for (let i = 0; i < len; i++) {
    for (let j = 0; j < needle.length; j++) {
      if (needle[j] !== haystack[j + i]) break;
      else if (j === needle.length - 1) return i;
    }
  }
  return -1;
};

console.log(strStr("hello", "ll"));
